cmd/promtool/main.go

```diff
@@ -20,43 +20,80 @@ import (
 	"path/filepath"
 	"strings"
 
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
 	yaml "gopkg.in/yaml.v2"
 
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/common/version"
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/pkg/rulefmt"
 	"github.com/prometheus/prometheus/promql"
-	"github.com/prometheus/prometheus/util/cli"
 	"github.com/prometheus/prometheus/util/promlint"
 )
 
-// CheckConfigCmd validates configuration files.
-func CheckConfigCmd(t cli.Term, args ...string) int {
-	if len(args) == 0 {
-		t.Infof("usage: promtool check-config <files>")
-		return 2
+func main() {
+	app := kingpin.New(filepath.Base(os.Args[0]), "Tooling for the Prometheus monitoring system.")
+	app.Version(version.Print("promtool"))
+	app.HelpFlag.Short('h')
+
+	checkCmd := app.Command("check", "Check the resources for validity.")
+
+	checkConfigCmd := checkCmd.Command("config", "Check if the config files are valid or not.")
+	configFiles := checkConfigCmd.Arg(
+		"config-files",
+		"The config files to check.",
+	).Required().ExistingFiles()
+
+	checkRulesCmd := checkCmd.Command("rules", "Check if the rule files are valid or not.")
+	ruleFiles := checkRulesCmd.Arg(
+		"rule-files",
+		"The rule files to check.",
+	).Required().ExistingFiles()
+
+	checkMetricsCmd := checkCmd.Command("metrics", checkMetricsUsage)
+
+	updateRulesCmd := app.Command("update-rules", "Update rules from the 1.x to 2.x format.")
+	ruleFilesUp := updateRulesCmd.Arg("rule-files", "The rule files to update.").Required().ExistingFiles()
+
+	switch kingpin.MustParse(app.Parse(os.Args[1:])) {
+	case checkConfigCmd.FullCommand():
+		CheckConfig(*configFiles...)
+
+	case checkRulesCmd.FullCommand():
+		CheckRules(*ruleFiles...)
+
+	case checkMetricsCmd.FullCommand():
+		os.Exit(CheckMetrics())
+
+	case updateRulesCmd.FullCommand():
+		UpdateRules(*ruleFilesUp...)
+
 	}
+
+}
+
+// CheckConfig validates configuration files.
+func CheckConfig(files ...string) int {
```

Comment:
Instead of returning an integer and printing to os.Stderr directly inside the function, what about returning the `error` and leave the printing + return code to the main function?